### PR TITLE
Add rate limit endpoint

### PR DESCRIPTION
**Pull request by quickliketurtle**
Hello, 
I've added the rate_limit endpoint. With the default response, as well as 2 helpers for core rate limit and search rate limit. 

I updated Client.php to allow both rateLimit and rate_limit options. 

I've added a Test with the sample rate_limit response from github, as well as a functional test ( the functional group is excluded now, but you can run the single file manually and it passes ). 

Also added a doc page.

Let me know if there are changed needed. I tried to follow style and naming convention by looking at existing code. 

Thanks.

### CODE REVIEW

File: lib/Github/Api/RateLimit.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ *
+ * @link   https://developer.github.com/v3/rate_limit/
+ * @author Jeff Finley <quickliketurtle@gmail.com>
+ */
+class RateLimit extends AbstractApi
+{
+    /**
+     * Get rate limits
+     *
+     * @return \Guzzle\Http\EntityBodyInterface|mixed|string
```
Comment (stof) [inline lib/Github/Api/RateLimit.php]: this looks weird. The other methods tell me that this returns an array.

Comment (quickliketurtle) [inline lib/Github/Api/RateLimit.php]: Ah, phpstrom thought for me on that one, will update to array.

**Comment by quickliketurtle** [comment]
Sorry for all the style fix commits... kept missing extra lines.

**Comment by Soullivaneuh** [comment]
:+1:

**Comment by stof** [comment]
your new doc page needs to be linked from other doc page (probably the doc index in this case)

**Comment by quickliketurtle** [comment]
@stof Good catch, i added the link to the readme.

File: lib/Github/Api/RateLimit.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    /**
+     * Get core rate limit
+     *
+     * @return integer
+     */
+    public function getCoreLimit()
+    {
+        $response = $this->getRateLimits();
+        return $response['resources']['core']['limit'];
```
Comment (cursedcoder) [inline lib/Github/Api/RateLimit.php]: please separate return statement with 1 blank line

Comment (Soullivaneuh) [inline lib/Github/Api/RateLimit.php]: @cursedcoder 

You should use `symfony` preset on your [StyleCI config file](https://github.com/KnpLabs/php-github-api/blob/a89996619661bc490a35ef5133defa87a4030e88/.styleci.yml) or at least, add the `return` fixer.

Comment (Soullivaneuh) [inline lib/Github/Api/RateLimit.php]: See https://github.com/KnpLabs/php-github-api/pull/323.

**Comment by cursedcoder** [comment]
@quickliketurtle very good, thank you!

**Pull request merged**